src/org/ods/services/OpenShiftService.groovy

```diff
@@ -171,4 +173,69 @@ class OpenShiftService {
       label: "Get phase of build ${buildId}"
     ).trim().toLowerCase()
   }
+  
+  public String getContainerForImage (String projectId, String rc, String image) {
+    script.sh(
+      script: """oc -n ${projectId} get rc ${rc} -o jsonpath='{.spec.template.spec.containers[?(contains .image "${image}")].name}'""",
+    )
+  }
+  
+  // Gets pod of deployment
+  Map getPodDataForDeployment(String componentId, String rc) {
+    
+    if (!automaticImageChangeTriggerEnabled(componentId)) {
```

Comment:
At least for the shared lib (not the MRO), we do not need an image trigger for deployments to work. If not image trigger is found, a manual rollout is made.

Reply:
this is for the tests also :) cool to leave in there?

Reply:
Not really I think. If we have it here, then we would need to remove the possibility to rollout without it. BUT I am not sure we want to do that. If it weren't for the MRO, I would actually recommend NOT to have any triggers. I am not a fan of this feature of the `DeploymentConfig` resource - plain Kubernetes `Deployment` resources don't have it ... but that is a bigger topic ...

One way to resolve this would be to expect triggers for `DeploymentConfig` resources, but to also support `Deployment` resources (which OpenShift does, and weirdly even recommends in https://docs.openshift.com/container-platform/4.1/applications/deployments/what-deployments-are.html#deployments-comparing-deploymentconfigs_what-deployments-are: "it is recommended to use Deployments unless you need a specific feature or behavior provided by DeploymentConfigs")